### ISSUE TITLE
Error message on module input type mismatch should not reference Go types

### ISSUE THREAD

**Issue by phinze** [hashicorp/terraform#7139]
### Terraform Version

master / 2127466280172a7cd0aa1d863cb932aa0107c986 

### Affected Resource(s)

core

### Terraform Configuration Files
```tf
# main.tf
variable "amap" {
  type    = "map"
  default = {  }
}
module "mod" {
  source   = "./mod"
  mapparam = "${var.amap}"
}
```

```tf
# mod/main.tf
variable "mapparam" { }
```

### Expected Behavior

An error message that does not reference Go types.

### Actual Behavior

```
Error configuring: 1 error(s) occurred:

* variable mapparam in module mod should be type string, got map[string]interface {}
```

**Comment by anosulchik** [comment]
+1

**Comment by phinze** [comment]
This has been fixed in ad27dc94fdd77e20b6cac4e9ba76db88aaed6e0a 👍 

New error:

```
* variable mapparam in module mod should be type string, got map
```
